Create Statement of Work after approving sponsorship + Send PDF document

berinhard opened the pull request:
This PR adds:

- Form to input data such as `start_date`, `end_date` to properly accept a sponsorship applications;
- Automatically creates a statement of work for the approved application;
- Statement of Work's admin management;
- SoW's preview
- Fix #1748

berinhard committed:
Prevents from rejecting or accepting reviewd applications

berinhard committed:
Display detailed sponsor's address information

berinhard committed:
Fix f-string and change postal code order

berinhard committed:
Model statement of work's information

berinhard committed:
Auto increment revision no

berinhard committed:
Automatically populate sponsor info and contact fields

berinhard committed:
Refactor SponsorBenefit creation

berinhard committed:
List benefits info

berinhard committed:
Format legal clauses list

berinhard committed:
Implement use case to approve sponsorship

berinhard committed:
Refactor use cases

berinhard committed:
Model admin for statements of work

berinhard committed:
Add link from approved sponsorship to draft SOW

berinhard committed:
Replace sponsorships preview by SoW preview

berinhard committed:
Black =]

berinhard committed:
Add button to preview SoW from change form

berinhard committed:
Move sponsor contact to sponsor info and display primary contact's email

berinhard committed:
Approve method requires start/end date

berinhard committed:
Accept use case now updates the sponsorship with more data

berinhard committed:
Display form when reviewing sponsorship application

berinhard committed:
Prevent sponsorship from being changed after approval/rejection

berinhard committed:
Black =S

berinhard committed:
Join markdowns so footnotes can work

berinhard committed:
Remove unecessary trailing spaces

berinhard committed:
Fix issue after rebase

berinhard committed:
Respect db ordering to avoid tests inconsistencies

berinhard committed:
Move admin views to a specific file

berinhard committed:
Install django easy pdf

berinhard committed:
Display SoW preview as PDF

berinhard committed:
Move django-easy-pdf code to a specific module

berinhard committed:
Refactor model's contants

berinhard committed:
Create auxiliar function to render PDF document as bytes

berinhard committed:
Add status control to Statement of Work model

berinhard committed:
Rename exception

berinhard committed:
Create function to save the final document version

berinhard committed:
Add vscode dir to gitignore

berinhard committed:
UC to send SoW

berinhard committed:
Update notifications to use SoW instead of sponsorship obj

berinhard committed:
Impleent view to send statement of work to users

berinhard committed:
Refactor to use EmailMessage instead of send_mail shortcut

berinhard committed:
Attach SoW PDF to emails

berinhard committed:
Add button to send SoW

berinhard committed:
Display an iframe with the PDF file before sending the document

berinhard committed:
Shouldn't edit document fields if not a draft version

berinhard committed:
Add administrative flag to sponsor's contact

berinhard committed:
Enable rollback sponsorship to edit

berinhard committed:
Admin view to rollback to edit

berinhard committed:
Add button in sponsorship's change form

berinhard committed:
Merge branch 'topic/edit-sponsorships' into feature/gen-statement-of-work

berinhard committed:
Move rollback view to views_admin to respect internals structure

berinhard committed:
Manage SoW before rolling back an application

berinhard committed:
Minor lint warnings

berinhard committed:
Merge branch 'master' into feature/gen-statement-of-work

# Conflicts:
#	sponsors/admin.py
#	sponsors/models.py
#	sponsors/tests/test_models.py
#	templates/sponsors/admin/sponsorship_change_form.html

berinhard committed:
Add document summary content

berinhard committed:
Style page to closer to the reference

berinhard committed:
Add contract bullet items

berinhard committed:
List benefits and legal clauses

berinhard committed:
Do not display legal clauses section if nothing to list

berinhard committed:
Merge branch 'master' into feature/gen-statement-of-work

# Conflicts:
#	sponsors/admin.py
#	sponsors/forms.py

berinhard committed:
Create merge migration

berinhard committed:
Remove element that was useful only for development

berinhard committed:
Fix pdf tests

berinhard committed:
Return 0 if sponsorship_fee is none None to avoid TypeError from num2words

ewdurbin commented on templates/sponsors/admin/preview-statement-of-work.html:
```diff
@@ -0,0 +1,283 @@
+and facilitate the growth of a diverse and international community of Python programmers (the
+<b>“Programs”</b>);</p>
+
+<p class='paragraph'><b>WHEREAS</b>, Sponsor is {{ sponsor.description }}; and</p>
```
I don't think we want the "description" here.

ewdurbin committed:
add missing configurations from #1735

berinhard committed:
Replace description text

berinhard committed:
Fix typo and add link

berinhard committed:
Rename StatementOfWork model

berinhard committed:
Update admin links

berinhard committed:
Rename sow variables to contract

berinhard committed:
Rename statement_of_work references

berinhard committed:
Rename last statement references

berinhard committed:
Move num2words requirement to base-requirements

berinhard committed:
Create commmand to work as initial data migration

berinhard committed:
Remove migration that doesn't work

berinhard committed:
Add logger to approve sponsorship use case

The logger is responsible to add a new entry in Django's LogEntry
to document the approval.

berinhard committed:
Remove expired todos

berinhard committed:
Add missing migration to replace SoW by Contract

berinhard committed:
Legal clauses can be empty

berinhard committed:
Also log actitivy when final contract is sent

ewdurbin committed:
Merge branch 'master' into feature/gen-statement-of-work

ewdurbin committed:
fix bad merge

ewdurbin committed:
Store original program name on SponsorBenefit objects

berinhard committed:
Benefits must always have a program

berinhard committed:
Prevent from sending contract file to sponsor

berinhard committed:
Implement logic to finalize contract and, thus, enable sponsorship benefits

berinhard committed:
Implement operation to nullify a contract
